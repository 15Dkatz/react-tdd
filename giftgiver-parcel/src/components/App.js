import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
import Gift from './Gift';
import { max_number } from '../helper';

class App extends Component {
  constructor() {
    super();

    this.state = { gifts: [] };
  }

  addGift = () => {
    const { gifts } = this.state;
    gifts.push({ id: max_number(this.state.gifts.map(gift => gift.id))+1 });
    this.setState({ gifts });
  }

  removeGift = id => {
    this.setState({ gifts: this.state.gifts.filter(gift => gift.id !== id) });
  }

  render() {
    return (
      <div>
        <h2>Gift Giver</h2>
        <div className='gift-list'>
          {
            this.state.gifts.map(gift => {
              return (
                <Gift
                  key={gift.id}
                  gift={gift}
                  removeGift={this.removeGift}
                />
              )
            })
          }
        </div>
        <Button className='btn-add' onClick={this.addGift}>Add Gift</Button>
      </div>
    )
  }
}

export default App;