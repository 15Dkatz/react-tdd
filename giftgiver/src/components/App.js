import React, { Component } from 'react';
import { Button } from 'react-bootstrap';

class App extends Component {
  constructor() {
    super();

    this.state = { gifts: [] };
  }

  addGift = () => {
    const { gifts } = this.state;

    const ids = this.state.gifts.map(gift => gift.id);

    const max_id = ids.length > 0 ? Math.max(...ids) : 0;

    gifts.push({ id: max_id+1 });

    this.setState({ gifts });
  }

  render() {
    return (
      <div>
        <h2>Gift Giver</h2>
        <Button className='btn-add' onClick={this.addGift}>Add Gift</Button>
      </div>
    )
  }
}

export default App;