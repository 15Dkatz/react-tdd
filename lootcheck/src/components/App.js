import React, { Component } from 'react';
import Wallet from './Wallet';

class App extends Component {
  render() {
    return (
      <div>
        <h2>Loot Check</h2>
        <hr />
        <Wallet />
      </div>
    )
  }
}

export default App;