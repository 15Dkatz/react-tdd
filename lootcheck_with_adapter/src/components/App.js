import React from 'react';
import Wallet from './Wallet';
import Loot from './Loot';

const App = () => {
  return (
    <div>
      <h2>Loot Check</h2>
      <hr />
      <Wallet />
      <hr />
      <Loot />
      <div>Powered by <a target='_blank' href='https://www.coindesk.com/price'>Coindesk</a></div>
    </div>
  )
}

export default App;