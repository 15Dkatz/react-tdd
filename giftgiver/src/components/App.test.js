import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

const app = shallow(<App />);

it('renders correctly', () => {
  expect(app).toMatchSnapshot();
});

it('initializes the `state` with an empty list of gifts', () => {
  expect(app.state().gifts).toEqual([]);
});

it('adds a new gift to `state` when clicking the `add gift` button', () => {
  app.find('.btn-add').simulate('click');

  expect(app.state().gifts).toEqual([{ id: 1 }]);
});