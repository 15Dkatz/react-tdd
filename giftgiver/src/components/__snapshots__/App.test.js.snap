// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h2>
        Gift Giver
    </h2>
    <div
        className="gift-list"
    >
        
    </div>
    <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="btn-add"
        disabled={false}
        onClick={[Function]}
    >
        Add Gift
    </Button>
</div>,
  "nodes": Array [
    <div>
      <h2>
            Gift Giver
      </h2>
      <div
            className="gift-list"
      >
            
      </div>
      <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            className="btn-add"
            disabled={false}
            onClick={[Function]}
      >
            Add Gift
      </Button>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "addGift": [Function],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "removeGift": [Function],
        "state": Object {
          "gifts": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h2>
                    Gift Giver
          </h2>
          <div
                    className="gift-list"
          >
                    
          </div>
          <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="btn-add"
                    disabled={false}
                    onClick={[Function]}
          >
                    Add Gift
          </Button>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h2>
                    Gift Giver
          </h2>
          <div
                    className="gift-list"
          >
                    
          </div>
          <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="btn-add"
                    disabled={false}
                    onClick={[Function]}
          >
                    Add Gift
          </Button>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
